### PR TITLE
Live blog auto refresh

### PR DESCRIPTION
**Pull request by gklopper**
Firstly apologies. This started out to fix a problem whereby people who left a tab open in the background with a live blog in it would eventually have performance difficulties due to constant updating. I fixed that by switching to the "there are more updates" mode.

Unfortunately along the way I was annoyed by random Javascript errors (mostly related to ads) causing parts of the live blogs to fail. 

I have always hated that one error in Javascript can take down everything else, so I decided to implement a plan I had for `robust()`, and one thing lead to the next.

Robust wraps everything in a `Promise` and swallows any exceptions (after logging them to Sentry). In "dev" mode it still logs to the console.

I wanted to try it out in Live Blogs, but the "perceived" performance improvement I saw on my machine made me put it in common too...

### CODE REVIEW

File: static/src/javascripts/projects/common/utils/robust.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        new Promise(function (resolve, reject) {
+            try {
+                block();
+                resolve();
+            } catch (e) {
+                reject(e);
+            }
+        }).catch(function (e) {
+            reporter(e, { tags: { module: name } });
+        });
```
Comment (ironsidevsquincy) [inline static/src/javascripts/projects/common/utils/robust.js]: what's the reason behind using a promise?

if `block` calls some async code, that wont be caught by the try/catch

File: common/app/views/fragments/javaScriptLaterSteps.scala.html [hunk 2]
```diff
@@ -146,6 +146,7 @@
                 'http://' + guardian.config.page.sentryPublicApiKey + '@@' + guardian.config.page.sentryHost,
                 {
                     whitelistUrls: [
+                        /localhost/,
```
Comment (ironsidevsquincy) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: don't think we want to send localhost errors to sentry

File: common/app/views/fragments/javaScriptLaterSteps.scala.html [hunk 3]
```diff
@@ -162,7 +163,11 @@
                         return data;
                     },
                     shouldSendCallback: function(data) {
-                        return Math.random() < 0.2 && guardian.config.switches.enableDiagnosticsLogging;
+                        @if(play.Play.isDev()) {
+                            console.error(data);
+                        }
+                                                                            @* don't actually notify sentry in dev mode*@
```
Comment (ironsidevsquincy) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: indentation

File: common/app/views/fragments/javaScriptLaterSteps.scala.html (hunk 2 above)
Comment (gklopper) [inline common/app/views/fragments/javaScriptLaterSteps.scala.html]: It does not. I will clean up the layout below to make that more obvious.

File: static/src/javascripts/projects/common/utils/robust.js (hunk 1 above)
Comment (gklopper) [inline static/src/javascripts/projects/common/utils/robust.js]: Good call. I started out actually intending to use the promises, but realised they were not exactly what I wanted. It did feel to me like the page rendered smoother with them in so I left them, but I ran some numbers this morning using the timing api and I don't think that is actually the case. Will replace with simpler `try { ... } catch (e) { ... }`.

File: sbt [hunk 4]
```diff
@@ -34,11 +34,9 @@ fi
 #http://stackoverflow.com/questions/5374455/what-does-java-option-xmx-stand-for
 jvm_mem="-Xmx$((physical_mem / 2 / 1024))m"
 
-#http://stackoverflow.com/questions/12114174/what-does-xxmaxpermsize-do
-perm_size="-XX:MaxPermSize=$((physical_mem / 3 / 1024))m"
```
Comment (gklopper) [inline sbt]: No longer used in JDK 8
http://stackoverflow.com/questions/18339707/permgen-elimination-in-jdk-8

**Comment by phamann** [comment]
:+1: Ship it!

**Comment by ironsidevsquincy** [comment]
Wonder if there's a less obtrusive way of doing this, e.g. hijack the `require` method to always wrap in a try/catch

Plus, note this won't catch errors thrown by ads, or outside of the same execution (e.g. in callbacks)

**Comment by gklopper** [comment]
@ironsidevsquincy am I correct in saying that errors in a different execution context will not cause the current execution context to stop? I don't mind if a module kills itself, I don't want it to kill the next module.

**Comment by ironsidevsquincy** [comment]
That's right

**Comment by gklopper** [comment]
I'm going to go with this. Lets see what it catches and then have a think about wrapping `require` of `define`.

**Pull request merged**